### PR TITLE
Ruowan/dependency

### PR DESCRIPTION
**Pull request by ruowan**

### CODE REVIEW

File: lib/commands/analyze-dependency.ts [hunk 1]
```diff
@@ -38,13 +41,18 @@ export const builder: yargs.CommandBuilder = {
...
+    let swaggerFilePaths: string[] = [];
+    if (argv.readme !== undefined) {
+      const readmeMd: string = argv.readme;
+      const autorestConfig = await getAutorestConfig(argv, readmeMd);
+      const fileRoot = path.dirname(readmeMd);
+      swaggerFilePaths = autorestConfig["input-file"].map((it: string) =>
+        path.resolve(fileRoot, it)
+      );
+    }
+    if (argv.swagger !== undefined) {
```
Comment (raych1) [inline lib/commands/analyze-dependency.ts:53]: Seems swagger input will take priority when both swagger and readme are specified. Can you add this information in the command line help output?

File: lib/testScenario/swaggerAnalyzer.ts [hunk 2]
```diff
@@ -245,9 +315,90 @@ export function analyzeExampleDependency(example: SwaggerExample): Dependency[]
...
+  }
+  return ret;
+}
+
+const uniqueIndex = (
+  jsonPointer: string,
+  swaggerFilePath: string,
+  resourceType: string
+): string => {
+  return `${jsonPointer}_${swaggerFilePath}_${resourceType}`;
```
Comment (raych1) [inline lib/testScenario/swaggerAnalyzer.ts:403]: Should we add repository, branch, and commit info?

File: lib/commands/analyze-dependency.ts (hunk 1 above)
Comment (ruowan) [reply lib/commands/analyze-dependency.ts:53]: Good suggestion.

File: lib/testScenario/swaggerAnalyzer.ts (hunk 2 above)
Comment (ruowan) [reply lib/testScenario/swaggerAnalyzer.ts:403]: No, we could add these infomation in our scripts.

File: test/testScenario/swaggerAnalyzerTest.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+      filerTopLevelResourceType: false,
+    });
+    await analyzer.initialize();
+    const exampleDependencies = await analyzer.analyzeDependency();
+    const normalizedResult = normalizeDependency(exampleDependencies);
+    expect(normalizedResult).toStrictEqual([
+      {
+        apiVersion: "2018-09-01-preview",
+        exampleName: "CreateOrUpdateApplication",
+        resourceType: "Microsoft.ServiceFabricMesh/applications",
```
[the remark this answers is not in the document]
Comment (ruowan) [reply test/testScenario/swaggerAnalyzerTest.ts:23]: will rename to fullResourceType

**Review by raych1 — APPROVED** [review]

**Review by PhoenixHe-msft — APPROVED** [review]

**Pull request merged**